Sources/SwiftParser/Types.swift

```diff
@@ -1045,7 +1045,21 @@ extension Parser {
         )
       )
     } else {
-      return self.parseType()
+      var result = self.parseType()
+      if self.at(.rightSquareBracket) {
+        let (unexpectedBeforeRSquareBracket, rightSquareBracket) = self.expect(.rightSquareBracket)
+        result = RawTypeSyntax(
+          RawArrayTypeSyntax(
+            leftSquareBracket: missingToken(.leftSquareBracket),
+            elementType: result,
+            unexpectedBeforeRSquareBracket,
+            rightSquareBracket: rightSquareBracket,
+            arena: self.arena
+          )
+        )
+      }
```

Comment:
I'm not sure if I'm doing this at the right place. 

I've also been looking into to putting this into `parseSimpleType`

Reply:
@ahoppen

Reply:
This is exactly the right place to handle it IMO. It’s also where the old parser handled it. https://github.com/apple/swift/blob/316dfaffa1a387dd0ab1b0e80f0e98c8cea09d7a/lib/Parse/ParseType.cpp#L646

In `parseSimpleType` you get into trouble because the `]` could belong to a subscript or an outer array type.

Reply:
We get an strange error with `"let a2: Set<Int1️⃣]>"`

The `var result = self.parseType()` return this tree. 
It look like it can't find the ending `>`

```
▿ Set<Int
  ▿ raw : .layout(simpleTypeIdentifier byteLength=7 descendantCount=8)
    - 0 : nil
    - 1 : .parsedToken(identifier wholeText="Set" textRange=0..<3)
    - 2 : nil
    ▿ 3 : .layout(genericArgumentClause byteLength=4 descendantCount=6)
      - 0 : nil
      - 1 : .parsedToken(leftAngle wholeText="<" textRange=0..<1)
      - 2 : nil
      ▿ 3 : .layout(genericArgumentList byteLength=3 descendantCount=3)
        ▿ 0 : .layout(genericArgument byteLength=3 descendantCount=2)
          - 0 : nil
          ▿ 1 : .layout(simpleTypeIdentifier byteLength=3 descendantCount=1)
            - 0 : nil
            - 1 : .parsedToken(identifier wholeText="Int" textRange=0..<3)
            - 2 : nil
            - 3 : nil
            - 4 : nil
          - 2 : nil
          - 3 : nil
          - 4 : nil
      - 4 : nil
      - 5 : .materializedToken(rightAngle text=">" numLeadingTrivia=0 byteLength=1)
      - 6 : nil
    - 4 : nil
```

Reply:
The fixed code ends look like `let a2: [Set<Int>]>`

Reply:
`"let a4: Int1️⃣]?"` also looks like it fails, because of the `?`. 
Should I try to refactor so we can reuse the optional or force unwrap inside `parseSimpleType`

Reply:
There wan't a way to look if a raw then had an error. 
So I've added one